src/fs/fs.c

```diff
@@ -545,19 +545,28 @@ int cfs_fs_delete(const str256_t *fs_name)
 		goto out;
 	}
 
+	/* Remove fs from the cortxfs list */
+	fs_node = container_of(fs, struct cfs_fs_node, cfs_fs);
+	LIST_REMOVE(fs_node, link);
+
+	/* Call fini for all members of fs */
 	RC_WRAP_LABEL(rc, out, cfs_ino_num_gen_fini, fs);
 	RC_WRAP_LABEL(rc, out, kvtree_fini, fs->kvtree);
+	kvnode_fini(fs->root_node);
```

Comment:
You need to do following 👍 
tenant_free
kvnode_fini
kvtree_fini
 free(fs_node->cfs_fs.ns);
 free(fs_node->cfs_fs.root_node);
 free(fs_node->cfs_fs.kvtree);
free(fs_node)

Reply:
kvnode_fini happens before kvtree_fini.
Few lines of this are going to be part of single function which I am working on. For the sake of this defect you can do this in order. I will change it to one line call as part of my changes.

If you have doubts, you can check with @pratyush-seagate , he has worked on this code before.